tests/tests_common.py

```diff
@@ -856,14 +882,57 @@ def test_method_array2(self):
     def test_add_nodes(self):
         objects = self.opc.get_objects_node()
         f = objects.add_folder(3, 'MyFolder')
+        child = objects.get_child("3:MyFolder")
+        self.assertEqual(child, f)
+        o = f.add_object(3, 'MyObject')
+        child = f.get_child("3:MyObject")
+        self.assertEqual(child, o)
         v = f.add_variable(3, 'MyVariable', 6)
+        child = f.get_child("3:MyVariable")
+        self.assertEqual(child, v)
         p = f.add_property(3, 'MyProperty', 10)
+        child = f.get_child("3:MyProperty")
+        self.assertEqual(child, p)
         childs = f.get_children()
+        self.assertTrue(o in childs)
         self.assertTrue(v in childs)
         self.assertTrue(p in childs)
 
+    def test_references_for_added_nodes(self):
+        objects = self.opc.get_objects_node()
+        o = objects.add_object(3, 'MyObject')
+        nodes = objects.get_referenced_nodes(refs=ua.ObjectIds.Organizes, direction=ua.BrowseDirection.Forward, includesubtypes=False)
+        self.assertTrue(o in nodes)
+        nodes = o.get_referenced_nodes(refs=ua.ObjectIds.Organizes, direction=ua.BrowseDirection.Inverse, includesubtypes=False)
+        self.assertTrue(objects in nodes)
+        self.assertEqual(o.get_parent(), objects)
+        self.assertEqual(o.get_type_definition(), ua.ObjectIds.BaseObjectType)
+
+        o2 = o.add_object(3, 'MySecondObject')
+        nodes = o.get_referenced_nodes(refs=ua.ObjectIds.HasComponent, direction=ua.BrowseDirection.Forward, includesubtypes=False)
+        self.assertTrue(o2 in nodes)
+        nodes = o2.get_referenced_nodes(refs=ua.ObjectIds.HasComponent, direction=ua.BrowseDirection.Inverse, includesubtypes=False)
+        self.assertTrue(o in nodes)
+        self.assertEqual(o2.get_parent(), o)
+        self.assertEqual(o2.get_type_definition(), ua.ObjectIds.BaseObjectType)
+
+        v = o.add_variable(3, 'MyVariable', 6)
+        nodes = o.get_referenced_nodes(refs=ua.ObjectIds.HasComponent, direction=ua.BrowseDirection.Forward, includesubtypes=False)
+        self.assertTrue(v in nodes)
+        nodes = v.get_referenced_nodes(refs=ua.ObjectIds.HasComponent, direction=ua.BrowseDirection.Inverse, includesubtypes=False)
+        self.assertTrue(o in nodes)
+        self.assertEqual(v.get_parent(), o)
+        self.assertEqual(v.get_type_definition(), ua.ObjectIds.BaseDataVariableType)
+
+        p = o.add_property(3, 'MyProperty', 2)
+        nodes = o.get_referenced_nodes(refs=ua.ObjectIds.HasProperty, direction=ua.BrowseDirection.Forward, includesubtypes=False)
+        self.assertTrue(p in nodes)
+        nodes = p.get_referenced_nodes(refs=ua.ObjectIds.HasProperty, direction=ua.BrowseDirection.Inverse, includesubtypes=False)
+        self.assertTrue(o in nodes)
+        self.assertEqual(p.get_parent(), o)
+        self.assertEqual(p.get_type_definition(), ua.ObjectIds.PropertyType)
+
```

Comment:
Yes I decided for the future to try to use Test-driven development, since it is easier to get what you want :)